doc/source/whatsnew/v0.25.0.rst

```diff
@@ -248,6 +248,7 @@ Performance Improvements
 - Improved performance of :meth:`pandas.core.groupby.GroupBy.quantile` (:issue:`20405`)
 - Improved performance of :meth:`read_csv` by faster tokenizing and faster parsing of small float numbers (:issue:`25784`)
 - Improved performance of :meth:`read_csv` by faster parsing of N/A and boolean values (:issue:`25804`)
+- Fixed performance degration of `DateTimeBlock`, by processing slicing trunk, not whole data. (:issue:`25708`)
```

Comment:
make this a user facing message, IOW this affects to_csv with datetime types

Reply:
@jreback done.